src/index.ts

```diff
@@ -83,6 +83,34 @@ class Trie<T> {
 
     return results;
   }
+
+  /**
+   * Returns an array of values that have keys matching the query exactly.
+   *
+   * You are encouraged to pass an options object with:
+   * a .limit to limit the number of results returned.
+   * a .unique property if you only want one result per-key.
```

Comment:
Is the `unique` property relevant for exact matches? How would one go about adding the same entry twice with the same key?

Reply:
The trie does not enforce unique keys. You can have multiple Andrew Lee contacts for example.

Reply:
Let me rephrase. Given that we're doing an exact match here, every key will necessarily be equivalent. Thus, isn't passing in `unique: true` equivalent to passing in `limit: 1`?

Reply:
I think that distinct is not useful here in **our** use, but the following is valid for the trie (duplicates can be inserted).

```
trie.add({key: 'andrew lee', distinct: 'andrew@a.com', score: 1});
trie.add({key: 'andrew lee', distinct: 'andrew@a.com', score: 2});
trie.add({key: 'andrew lee', distinct: 'drew@b.com', score: 3});
trie.add({key: 'andrew lee', distinct: 'lee@c.com', score: 4});
```

trie.{exact,prefix}Search({key: 'andrew lee', distinct: true}) => 3 results;
trie.{exact,prefix}Search({key: 'andrew lee', distinct: false}) => 4 results;

Reply:
Ah, I misunderstood the naming conventions here.

- `distinct` is an optional field on _every item_.
- `key` is overloaded to mean both "the string that is used to look up this item" and "this item's `distinct` field, if present, else its normal key"
- `unique` is a search-time option to deduplicate by `key` in the second sense

When the documentation for this method says

>   * a .unique property if you only want one result per-key.

I was interpreting it to mean "one result per string used to look up this item". Indeed, if you do not provide a `distinct` field for your items, the `unique` option seems pretty useless for an exact search.

Reply:
Even if it's just informally, I think we should have distinct words for `node.key` and `node.distinct || node.key`. I'd propose "key" and "identifier".

Reply:
> Indeed, if you do not provide a distinct field for your items, the unique option seems pretty useless for an exact search.

This is not true because the trie allows duplicate keys and values. So distinct can give you the same value multiple times.

FWIW we don't use/need any of this so we can simplify the interface. I don't really care but I request to review the PRs if we do want to change them.

Reply:
Talked out-of-band to resolve this. The documentation is imprecise and we can probably improve it, but the core insight is that `unique` operates on the `distinct` field, and is thus non-trivial.